src/app/bios.rs

```diff
@@ -478,3 +487,84 @@ impl Component for BiosComponent {
         Ok(())
     }
 }
+
+fn compact_smmstore(data: &[u8]) -> Vec<u8> {
```

Comment:
It would be good if this came from firmware-smmstore as a library

Reply:
Right, that's what I'm working on today. Also want to do something with [system76/smmstore](https://github.com/), since it has the same logic. It could probably just be added as an example in firmware-smmstore.